src/main/java/org/jabref/gui/preferences/customization/CustomizationTabViewModel.java

```diff
@@ -45,4 +108,58 @@ public BooleanProperty useCustomDOIProperty() {
     public StringProperty useCustomDOINameProperty() {
         return this.useCustomDOINameProperty;
     }
+
+    public ListProperty<CustomApiKeyPreferences> customApiKeyPrefsProperty() {
+        return this.customApiKeyPrefsListProperty;
+    }
+
+    public ObjectProperty<CustomApiKeyPreferences> selectedCustomApiKeyPrefProperty() {
+        return this.selectedCustomApiKeyPrefProperty;
+    }
+
+    public BooleanProperty useCustomApiKeyProperty() {
+        return this.useCustomApiKeyProperty;
+    }
+
+    public StringProperty customApiKeyText() {
+        return this.customApiKeyTextProperty;
+    }
+
+    public void checkCustomApiKey() {
+        final String apiKeyName = selectedCustomApiKeyPrefProperty.get().getName();
+        final String testUrlWithoutApiKey = API_KEY_NAME_URL.get(apiKeyName);
+        final String apiKey = customApiKeyTextProperty.get();
+
+        final String connectionSuccessText = Localization.lang("Connection successful!");
+        final String connectionFailedText = Localization.lang("Connection failed!");
+        final String dialogTitle = Localization.lang("Check %0 API Key Setting", apiKeyName);
+
+        boolean valid;
+        if (!apiKey.isEmpty()) {
+            URLDownload urlDownload;
+            try {
+                SSLSocketFactory defaultSslSocketFactory = HttpsURLConnection.getDefaultSSLSocketFactory();
+                HostnameVerifier defaultHostnameVerifier = HttpsURLConnection.getDefaultHostnameVerifier();
+                URLDownload.bypassSSLVerification();
+
+                urlDownload = new URLDownload(testUrlWithoutApiKey + apiKey);
+                // The HEAD request cannot be used because its response is not 200 (maybe 404 or 596...).
+                int statusCode = ((HttpURLConnection) urlDownload.getSource().openConnection()).getResponseCode();
+                valid = statusCode >= 200 && statusCode < 300;
+
+                URLDownload.setSSLVerification(defaultSslSocketFactory, defaultHostnameVerifier);
+            } catch (IOException | kong.unirest.UnirestException e) {
+                valid = false;
+            }
+        } else {
+            valid = false;
+        }
+
+        if (valid) {
+            dialogService.showInformationDialogAndWait(dialogTitle, connectionSuccessText);
+        } else {
+            dialogService.showErrorDialogAndWait(dialogTitle, connectionFailedText);
+        }
```

Comment:
On very simple distinctions I would say, that you can use the quick'n'dirty `?` operator. This would mean here:
`dialogService.showInformationDialogAndWait(dialogTitle, valid ? connectionSuccessText : connectionFailedText);`
But i'm not sure about our policy in JabRef right now. Anybody @JabRef/developers ?

Reply:
If the connection test fails, should you use the error dialog or the information dialog?